File: backend/app/schemas/store.py
import uuid
from typing import Literal

from pydantic import BaseModel


STORE_PLATFORMS = Literal["steam", "playstation", "nintendo", "xbox", "epic", "gog"]


class StoreLink(BaseModel):
    platform: STORE_PLATFORMS
    url: str


class CitedGame(BaseModel):
    id: uuid.UUID
    title: str
    cover_url: str | None = None
    store_links: list[StoreLink] = []
    platforms: list[str] = []

File: frontend/src/types/store.ts
import { z } from 'zod'

export const STORE_PLATFORMS = ['steam', 'playstation', 'nintendo', 'xbox', 'epic', 'gog'] as const
export type StorePlatform = (typeof STORE_PLATFORMS)[number]

export const StoreLinkSchema = z.object({
  platform: z.enum(STORE_PLATFORMS),
  url: z.string(),
})

export const CitedGameSchema = z.object({
  id: z.string(),
  title: z.string(),
  cover_url: z.string().nullable(),
  store_links: z.array(StoreLinkSchema).default([]),
  platforms: z.array(z.string()).default([]),
})

export type StoreLink = z.infer<typeof StoreLinkSchema>
export type CitedGame = z.infer<typeof CitedGameSchema>
